### DISCUSSION TITLE
Expose the cache to users and extend its capabilties

### DISCUSSION THREAD

**Discussion by seladb** [litestar-org/litestar#606] [Ideas]
We started discussing it here: https://github.com/starlite-api/starlite/issues/534, but it's probably worth its own thread.

To the best of my knowledge, Starlite currently uses the cache for internal purposes such as caching responses and rate limiting (let me know if I'm missing something). However, the cache is already exposed to users through [`app.cache`](https://github.com/starlite-api/starlite/blob/ac5f92935bbbb9959bf92650351eec3375a04784/starlite/app.py#L332), so I'm wondering if we should extend the cache capabilities and let users use it for any purpose. We can use [Django’s cache framework](https://docs.djangoproject.com/en/4.1/topics/cache/) as an inspiration.

If we want to do that we should also think how to expose it better in route handlers and controllers.

Please let me know what you think. Maybe it is out-of-scope for Starlite, which I can totally understand, and we should close this discussion. I'm just thinking that for many real-life applications, cache and DB are pretty basic infrastructure, and if Starlite facilitates them it can reduce the number of integrations and complexity for users to build their applications.

**Comment by Goldziher** [comment]
So, cache is available in route handlers via `request.cache` and `socket.cache`. It's also accessible in Middleware either via these, or by accessing `scope["app"].cache`. 

As for extending the protocol and `Cache` class - we can definitely do it. It's important though to keep bloat to a minimum and avoid breaking changes.

**Reply by seladb** [reply]
Ok sure, I'll keep it in my backlog and work on it sometime soon 👍

**Comment by provinzkraut** [comment]
Implemented in #1184 and #1330
